src/components/tooltip/tooltip-test.stories.tsx

```diff
@@ -0,0 +1,202 @@
+      options: ["left", "right", "top", "bottom"],
+      control: {
+        type: "select",
+      },
+    },
+    position: {
+      options: TOOLTIP_POSITIONS,
+      control: {
+        type: "select",
+      },
+    },
+    bgColor: {
+      control: {
+        type: "text",
+      },
+    },
+    fontColor: {
+      control: {
+        type: "text",
+      },
+    },
+  },
+};
+
+interface TooltipStoryProps {
+  message?: React.ReactNode;
+  isVisible?: boolean;
+  flipOverrides?: TooltipPositions;
+}
+
+export const DefaultStory = ({
+  message,
+  isVisible,
+  flipOverrides,
+  ...args
+}: TooltipStoryProps) => {
+  const [stateVisible, setStateVisible] = useState(false);
+  return (
+    <div
+      style={{
+        margin: "200px 400px",
+        width: 350,
+        height: 250,
+        overflow: "auto",
+      }}
+    >
+      <Tooltip
+        isVisible={stateVisible || isVisible}
+        flipOverrides={flipOverrides && [flipOverrides]}
+        message={message}
+        {...args}
+      >
+        <div
+          style={{
+            backgroundColor: "#00815D",
+            color: "white",
+            width: "42px",
+            height: "26px",
+            position: "relative",
+            outline: "none",
+            cursor: "pointer",
+          }}
+          {...(isVisible === false && {
+            onMouseOver: () => setStateVisible(true),
+            onMouseLeave: () => setStateVisible(false),
+            onFocus: () => setStateVisible(true),
+            onBlur: () => setStateVisible(false),
+          })}
+        >
+          <span style={{ position: "absolute", top: "4px", left: "2px" }}>
+            target
+          </span>
+        </div>
+      </Tooltip>
+    </div>
+  );
+};
+
+DefaultStory.story = {
+  name: "default",
+  args: {
+    position: "top",
+    isVisible: true,
+    message:
+      "I'm a helpful tooltip that can display additional information to a user.",
+    type: "default",
+    size: "medium",
+    bgColor: undefined,
+    fontColor: undefined,
+  },
+};
+
+const Button = forwardRef<HTMLButtonElement, ButtonProps>(
+  ({ children }, ref) => (
+    <button
+      type="button"
+      tabIndex={0}
+      data-component="tooltip-trigger"
+      style={{
+        backgroundColor: "#00815D",
+        color: "white",
+        cursor: "pointer",
+        border: "none",
+        padding: "8px",
+      }}
+      ref={ref}
+    >
+      {children}
+    </button>
+  )
+);
+Button.displayName = "Tooltip";
+
+const SecondaryButton = forwardRef<HTMLButtonElement, ButtonProps>(
+  ({ children }, ref) => (
+    <button
+      type="button"
+      tabIndex={0}
+      data-component="tooltip-trigger"
+      ref={ref}
+    >
+      {children}
+    </button>
+  )
+);
+SecondaryButton.displayName = "Tooltip";
+
+export const TooltipComponent = ({ ...props }) => (
+  <div
+    style={{
+      padding: "60px 60px 60px 160px",
+    }}
+  >
+    <Tooltip message="I am a tooltip!" isVisible {...props}>
```

Comment:
Similarly here, we need to type the incoming props as `Tooltip` props here.

```suggestion
export const TooltipComponent = ({ message, ...rest }: TooltipProps) => (
  <div
    style={{
      padding: "60px 60px 60px 160px",
    }}
  >
    <Tooltip message={message || "I am a tooltip!"} isVisible {...props}>
```

Since `message` can be passed but has a default value of "I am tooltip", we also need the ability to override its value if `message` has passed to `TooltipComponent` from the cypress tests.

Reply:
Should line 163 be:
<Tooltip message="I am a tooltip!" isVisible {...rest}>